packages/shared/lib/sdk/sync.ts

```diff
@@ -307,6 +339,173 @@ export class NangoAction {
 
         return this.attributes as A;
     }
+
+    public async *paginate<T = any>(config: ProxyConfiguration): AsyncGenerator<T[], undefined, void> {
+        const providerConfigKey: string = this.providerConfigKey as string;
+        const template: Template = configService.getTemplate(providerConfigKey);
+        const templatePaginationConfig: Pagination | undefined = template.proxy?.paginate;
+
+        if (!templatePaginationConfig) {
+            throw Error(`Pagination is not supported for '${providerConfigKey}'. Please, add pagination config to 'providers.yaml' file`);
```

Comment:
Can't we suggest that they provide an override here too?

Adding things to `providers.yaml` is heavy and requires a code review. There's little chance they'll do it.

Reply:
I have put this on my follow-up list.

Letting them do this is my ideal vision but I feel a bit hesitant about this change since there is no runtime type validation so I wanted to postpone this till we introduce something like `zod`. Basically, if I were to remove this check, the user won't really receive a useful error message from us if they supply a wrong/partial pagination config. This affects DX and potentially slightly increases support volume.

That said, I introduced this check back when there was no compile time check of the pagination config override so it should be safer now as we have `Partial<CursorPagination> | Partial<NextUrlPagination> | Partial<OffsetPagination>` check in place.

The tradeoff here is that we will temporarily force users to create PRs to update `providers.yaml` but reduce the chance of error.

I'm starting to lean toward removing this check. @khaliqgant @rguldener Any thoughts on this?

Reply:
Wait a second, PR won't be enough. They will need to wait till the release of this change, right? I'm leaning toward opening this API even more now.

Reply:
> override the pagination config in code.

The current check doesn't really care if there is an override in the code or not. It throws anyway if it sees that there is no default config in `providers.yaml`. That's why I mentioned that we will have to update it to throw an error if there is no default config and the user didn't provide overrides.